File: results/transcoder-st/python/semantics/MINIMUM_ROTATIONS_UNLOCK_CIRCULAR_LOCK.cpp
// Copyright (c) 2019-present, Facebook, Inc.
// All rights reserved.
//
// This source code is licensed under the license found in the
// LICENSE file in the root directory of this source tree.
//

#include <iostream>
#include <cstdlib>
#include <string>
#include <vector>
#include <fstream>
#include <iomanip>
#include <vector>
#include <map>
#include <unordered_map>
#include <set>
#include <unordered_set>
#include <numeric>
#include <algorithm>
using namespace std;
int f_gold ( int input, int unlock_code ) {
  int rotation = 0;
  int input_digit, code_digit;
  while ( input || unlock_code ) {
    input_digit = input % 10;
    code_digit = unlock_code % 10;
    rotation += min ( abs ( input_digit - code_digit ), 10 - abs ( input_digit - code_digit ) );
    input /= 10;
    unlock_code /= 10;
  }
  return rotation;
}


int f_filled ( int input , int unlockCode ) {
  int rotation = 0 ;
  ;
  while ( ( input > 0 || unlockCode > 0 ) && ( input < 0 || unlockCode < 0 ) ) {
    int inputDigit = input % 10 ;
    int codeDigit = unlockCode % 10 ;
    rotation += min ( abs ( inputDigit - codeDigit ) , 10 - abs ( inputDigit - codeDigit ) ) ;
    input = ( int ) ( input / 10 ) ;
    unlockCode = ( int ) ( unlockCode / 10 ) ;
  }
  ;
  return rotation ;
}


int main() {
    int n_success = 0;
    vector<int> param0 {71,90,28,41,32,39,33,89,50,92};
    vector<int> param1 {46,65,84,23,58,82,58,32,51,77};
    for(int i = 0; i < param0.size(); ++i)
    {
        if(f_filled(param0[i],param1[i]) == f_gold(param0[i],param1[i]))
        {
            n_success+=1;
        }
    }
    cout << "#Results:" << " " << n_success << ", " << param0.size();
    return 0;
}

File: traces/pdb_tests/traces_old/MINIMUM_ROTATIONS_UNLOCK_CIRCULAR_LOCK.py
Statement: rotation = 0 ;
Line: 8
input = 71
unlock_code = 46

Statement: while ( input > 0 or unlock_code > 0 ) :
Line: 9
input = 71
unlock_code = 46
rotation = 0

Statement: input_digit = input % 10 ;
Line: 0
input = 71
unlock_code = 46
rotation = 0

Statement: code_digit = unlock_code % 10 ;
Line: 1
input = 71
unlock_code = 46
rotation = 0
input_digit = 1

Statement: rotation += min ( abs ( input_digit - code_digit ) , 10 - abs ( input_digit - code_digit ) ) ;
Line: 2
input = 71
unlock_code = 46
rotation = 0
input_digit = 1
code_digit = 6

Statement: input = int ( input / 10 ) ;
Line: 3
input = 71
unlock_code = 46
rotation = 5
input_digit = 1
code_digit = 6

Statement: unlock_code = int ( unlock_code / 10 ) ;
Line: 4
input = 7
unlock_code = 46
rotation = 5
input_digit = 1
code_digit = 6

Statement: while ( input > 0 or unlock_code > 0 ) :
Line: 9
input = 7
unlock_code = 4
rotation = 5
input_digit = 1
code_digit = 6

Statement: input_digit = input % 10 ;
Line: 0
input = 7
unlock_code = 4
rotation = 5
input_digit = 1
code_digit = 6

Statement: code_digit = unlock_code % 10 ;
Line: 1
input = 7
unlock_code = 4
rotation = 5
input_digit = 7
code_digit = 6

Statement: rotation += min ( abs ( input_digit - code_digit ) , 10 - abs ( input_digit - code_digit ) ) ;
Line: 2
input = 7
unlock_code = 4
rotation = 5
input_digit = 7
code_digit = 4

Statement: input = int ( input / 10 ) ;
Line: 3
input = 7
unlock_code = 4
rotation = 8
input_digit = 7
code_digit = 4

Statement: unlock_code = int ( unlock_code / 10 ) ;
Line: 4
input = 0
unlock_code = 4
rotation = 8
input_digit = 7
code_digit = 4

Statement: while ( input > 0 or unlock_code > 0 ) :
Line: 9
input = 0
unlock_code = 0
rotation = 8
input_digit = 7
code_digit = 4

Statement: return rotation ;
Line: 5
input = 0
unlock_code = 0
rotation = 8
input_digit = 7
code_digit = 4

Statement: return rotation ;
Line: 5
input = 0
unlock_code = 0
rotation = 8
input_digit = 7
code_digit = 4
__return__ = 8
